### ISSUE TITLE
Router not passing on query parameters

### ISSUE THREAD

**Issue by btaens** [ratchetphp/Ratchet#367]
Hello,
I use a `_token` query parameter to authenticate my users on connection.
This works fine, as long as I don't want separate routes for my WebSocket apps.
The `\Ratchet\Http\Router` class seems to do this:

``` php
$parameters = array();
foreach($route as $key => $value) {
    if ((is_string($key)) && ('_' !== substr($key, 0, 1))) {
        $parameters[$key] = $value;
    }
}
$url = Url::factory($request->getPath());
$url->setQuery($parameters);
$request->setUrl($url);
```

This however, returns an empty array, and loses my `_token` parameter.

Simply rewriting it to this solves the problem:

``` php
$url = Url::factory($request->getPath());
$url->setQuery($request->getQuery());
$request->setUrl($url);
```

Though I might be missing something, maybe it breaks something else.

**Comment by cboden** [comment]
I would recommend changing your parameter to `token`. IIRC the `_` are omitted because those are the reserved parameters used to configure Symfony controls.

**Comment by btaens** [comment]
I tried that as well, I did notice it doing something like that, but that doesn't seem to be the problem.
It couldn't detect it even after renaming it to `token`.

**Comment by cboden** [comment]
This may have been fixed in #471
